lexos/models/stats_model.py

```diff
@@ -0,0 +1,179 @@
+    id_temp_label_map: IdTempLabelMap
+
+
+class CorpusInfo(NamedTuple):
+    """A typed tuple to represent statistics of the whole corpus."""
+    q1: float  # The first quartile of all file sizes.
+    q3: float  # The second quartile of all file sizes.
+    iqr: float  # The inter-quartile range of all file sizes.
+    median: float  # The median of all file sizes.
+    average: float  # The average of all file sized.
+    num_file: int  # The number of files.
+    file_sizes: list  # The list of all file sizes.
+    file_names: list  # The list of all file names.
+    std_deviation: float  # The standard deviation of all file sizes.
+    anomaly_iqr: dict  # The anomaly inter-quartile range of all file sizes.
+    anomaly_std_err: dict  # The anomaly standard error of all file sizes.
+
+
+class FileInfo(NamedTuple):
+    """A typed tuple to represent statistics of each file in corpus."""
+    q1: float  # The first quartile of all word counts of a file.
+    q3: float  # The second quartile of all word counts of a file.
+    iqr: float  # The inter-quartile range of all word counts of a file.
+    hapax: int  # The hapax of all word counts of a file.
+    median: float  # The median of all word counts of a file.
+    average: float  # The average of all word counts of a file.
+    num_word: int  # The number of words of a file.
+    file_name: str  # The name of a file.
+    word_count: int  # The count of all words within a file.
+    std_deviation: float  # The standard deviation of word counts.
+    total_word_count: int  # The total of all word counts of a file.
+
+
+class StatsModel(BaseModel):
+    def __init__(self, test_options: Optional[StatsTestOptions] = None):
+        """This is the class to generate statistics of the input file.
+
+        :param test_options: the input used in testing to override the
+                             dynamically loaded option
+        """
+        super().__init__()
+        if test_options is not None:
+            self._test_dtm = test_options.doc_term_matrix
+            self._test_id_temp_label_map = test_options.id_temp_label_map
+        else:
+            self._test_dtm = None
+            self._test_id_temp_label_map = None
+
+    @property
+    def _doc_term_matrix(self) -> pd.DataFrame:
+        """:return: the document term matrix"""
+        return self._test_dtm if self._test_dtm is not None \
+            else MatrixModel().get_matrix()
+
+    @property
+    def _id_temp_label_map(self) -> IdTempLabelMap:
+        """:return: a map takes an id to temp labels"""
+        return self._test_id_temp_label_map \
+            if self._test_id_temp_label_map is not None \
+            else MatrixModel().get_temp_label_id_map()
+
+    def _get_corpus_info(self) -> CorpusInfo:
+        """Converts word lists completely to statistic."""
+        assert np.sum(self._doc_term_matrix.values) > 0, EMPTY_LIST_MESSAGE
+        # initialize
+        file_anomaly_iqr = {}
```

Comment:
Please explain the stats you are calculating a little bit. As someone who has never worked on this functionality, I am lost!